comment/mixins.py

```diff
@@ -29,6 +32,39 @@ class BaseCommentMixin(LoginRequiredMixin, BasePermission):
     pass
 
 
+class CommentCreateMixin:
+    email_service = None
+
+    def _initialize_email_service(self, comment, request):
+        self.email_service = DABEmailService(comment, request)
+
+    def send_notification_to_followers(self, comment, request):
+        if settings.COMMENT_ALLOW_SUBSCRIPTION:
+            self._initialize_email_service(comment, request)
```

Comment:
This didn't solve the issue, calling the method here is still dplicating the code and creating 2 email service instances. `initialize_email_service` should be called **once** inside the view that inherits this mixin.

Reply:
I see that `send_notification_to_followers` is always called after `perform_save` so it can be called inside `perform_save`.

Reply:
> This didn't solve the issue, calling the method here is still dplicating the code and creating 2 email service instances. initialize_email_service should be called once inside the view that inherits this mixin.

```python
if settings.COMMENT_ALLOW_ANONYMOUS and not comment.user:
        self._handle_anonymous(comment, request, api)
else:
       self.perform_save(comment)
       self.send_notification_to_followers(comment, request)
return comment
```

Only one of the two branches can be executed at a time. I don't see how 2 instances of the email service are created.


>

Reply:
yes you are right. However moving `send_notification_to_followers` to `perform_save` is more cleaner.

Reply:
> I see that send_notification_to_followers is always called after perform_save so it can be called inside perform_save.

nice observation